src/splunkeventsender.py

```diff
@@ -0,0 +1,99 @@
+# the following allows JSON to be posted to Splunk HEC (HTTP Event Collector)
+
+import logging
+import os
+import json
+from src.kms import decrypt
+
+
+class SplunkEventSender:
+    def __init__(
+        self,
+        hec_token,
+        hec_host,
+        index="main",
+        host=None,
+        source=None,
+        sourcetype="json",
+        hec_port='443'
+    ):
+        self.index = index
+        self.host = host
+        self.source = source
+        self.sourcetype = sourcetype
+        self.hec_token = hec_token
+        self.uri = f"https://{hec_host}:{hec_port}/services/collector"
+
+    def base_json_event(self):
+        return {
+            "host": self.host,
+            "source": self.source,
+            "sourcetype": self.sourcetype,
+            "index": self.index,
+            "event": {},
+        }
+
+    def send(self, payload):
+        logger = logging.getLogger('splunk-event-send')
+        logger.setLevel(logging.INFO)
+
+        headers = {
+            'Authorization': f'Splunk {self.hec_token}',
+            'Content-Type': 'application/json'
+        }
+
+        if isinstance(payload, str):
+            payload = json.loads(payload)
+
+        json_event = self.base_json_event()
+
+        if "timestamp" in payload:
+            json_event["time"] = payload["timestamp"]
+
+        json_event["event"] = json.dumps(payload)
+
+        event = str(json.dumps(json_event))
+
+        try:
+            r = requests.post(self.uri, data=event, headers=headers, verify=True)
+            res = r.status_code, r.text,
+        except Exception as e:
+            logger.error("SplunkEventSender:send:", e)
+            res = False
+        return res
+
+
+def push_event_to_splunk(event):
+    logger = logging.getLogger('push-event-to-splunk')
+    logger.setLevel(logging.INFO)
+
+    if 'production' in os.environ['QUEUE_URL'] 
```

Comment:
```suggestion
    if 'production' in os.environ['QUEUE_URL']:
```

Reply:
Also, I feel this a little bit of a haphazard way of deciding whether to send to Splunk ... would having an explicit environment variable, e.g. SEND_TO_SPLUNK, that if present and set would result in things being sent!?